docs/samples/Microsoft.ML.Samples/Dynamic/MatrixFactorization.cs

```diff
@@ -0,0 +1,102 @@
+            // Contieuous=true means that all values between the min and max indexes are all allowed.
+            [KeyType(Contiguous = true, Count = _synthesizedMatrixColumnCount, Min = _synthesizedMatrixFirstColumnIndex)]
+            public uint MatrixColumnIndex;
+            // Matrix row index starts from _synthesizedMatrixFirstRowIndex and is at most
+            // _synthesizedMatrixFirstRowIndex + _synthesizedMatrixRowCount - 1.
+            // Contieuous=true means that all values between the min and max indexes are all allowed.
+            [KeyType(Contiguous = true, Count = _synthesizedMatrixRowCount, Min = _synthesizedMatrixFirstRowIndex)]
+            public uint MatrixRowIndex;
+            // The value at the column MatrixColumnIndex and row MatrixRowIndexin.
+            public float Value;
+        }
+
+        // A data structure used to encode prediction result. Comparing with MatrixElement, The field Value in MatrixElement is
+        // renamed to Score because Score is the default name of matrix factorization's output.
+        internal class MatrixElementForScore
+        {
+            [KeyType(Contiguous = true, Count = _synthesizedMatrixColumnCount, Min = _synthesizedMatrixFirstColumnIndex)]
+            public uint MatrixColumnIndex;
+            [KeyType(Contiguous = true, Count = _synthesizedMatrixRowCount, Min = _synthesizedMatrixFirstRowIndex)]
+            public uint MatrixRowIndex;
+            public float Score;
+        }
+
+        // This example first creates in-memory data and then use it to train a matrix factorization model. Afterward, quality metrics are reported.
+        public static void MatrixFactorizationInMemoryData()
+        {
+            // Create an in-memory matrix as a list of tuples (column index, row index, value).
+            var dataMatrix = new List<MatrixElement>();
+            for (uint i = _synthesizedMatrixFirstColumnIndex; i < _synthesizedMatrixFirstColumnIndex + _synthesizedMatrixColumnCount; ++i)
+                for (uint j = _synthesizedMatrixFirstRowIndex; j < _synthesizedMatrixFirstRowIndex + _synthesizedMatrixRowCount; ++j)
+                    dataMatrix.Add(new MatrixElement() { MatrixColumnIndex = i, MatrixRowIndex = j, Value = (i + j) % 5 });
+
+            // Creating the ML.Net IHostEnvironment object, needed for the pipeline
+            var mlContext = new MLContext(seed: 0, conc: 1);
+
+            // Convert the in-memory matrix into an IDataView so that ML.NET components can consume it.
+            var dataView = ComponentCreation.CreateDataView(mlContext, dataMatrix);
+
+            // Create a matrix factorization trainer which may consume "Value" as the training label, "MatrixColumnIndex" as the
+            // matrix's column index, and "MatrixRowIndex" as the matrix's row index.
+            var pipeline = new MatrixFactorizationTrainer(mlContext, "Value", "MatrixColumnIndex", "MatrixRowIndex",
```

Comment:
>"Value" [](start = 69, length = 7)

can you do nameof(MatrixElement.Value), nameof(MatrixElement.MatrixColumnIndex) and nameof(MatrixElement.MatrixRowIndex)?

Reply:
Ok.

---
In reply to: [229810285](https://github.com/dotnet/machinelearning/pull/1450#discussion_r229810285) [](ancestors = 229810285)